requirements: Per recommendation, constrain version to <0.15

jcfr opened the pull request:
> This is the new (0.15+) interface for ``ruamel.yaml``, it is still in
> the process of being fleshed out. Please pin your dependency to
> ruamel.yaml<0.15 for production software.

Source: http://yaml.readthedocs.io/en/latest/basicuse.html

nicain commented on setup.py:
```diff
@@ -27,7 +27,7 @@
         'numpy',
         'h5py',
-        'ruamel.yaml',
+        'ruamel.yaml<15',
```
Just double checking that this should be 15 and not .15

jcfr replied:
Good catch.

nicain reviewed (commented):
I really like using -e and -U before pip freeze

jcfr commented:
Since all test are currently passing,  and this addresses no particular problem. I am closing it.

I originally thought this was related to #248, but it is not.

@nicain @ajtritt @oruebel I think differently, we can still re-open and integrate.

jcfr committed:
requirements: Per recommendation, constrain ruamel.yaml version to <0.15


> This is the new (0.15+) interface for ``ruamel.yaml``, it is still in
> the process of being fleshed out. Please pin your dependency to
> ruamel.yaml<0.15 for production software.

Source: http://yaml.readthedocs.io/en/latest/basicuse.html
